external-crates/move/crates/move-compiler/src/cfgir/translate.rs

```diff
@@ -95,28 +86,32 @@ impl<'env> Context<'env> {
                 loop_end: G::LoopEnd::Target(end_label),
             },
         );
-        // push a new loop env
-        let old_env = mem::take(&mut self.loop_env);
-        self.loop_env = Some(LoopEnv {
-            start_label,
-            end_label,
-            previous_env: Box::new(old_env),
-        });
+        self.named_blocks
+            .add(name, (start_label, end_label))
+            .expect("ICE reused block name");
```

Comment:
Do they get gensymed before this?

Reply:
Yes, they are uniquely renamed during `naming`.